File: argentum-api/src/main/java/net/devium/argentum/rest/model/response/OrderItemResponse.java
package net.devium.argentum.rest.model.response;

import net.devium.argentum.jpa.OrderItemEntity;

public class OrderItemResponse {
    private final long id;
    private final long product;
    private final int quantity;
    private final int cancelled;

    private OrderItemResponse(long id, long product, int quantity, int cancelled) {
        this.id = id;
        this.product = product;
        this.quantity = quantity;
        this.cancelled = cancelled;
    }

    public static OrderItemResponse from(OrderItemEntity orderItem) {
        return new OrderItemResponse(
                orderItem.getId(),
                orderItem.getProduct().getId(),
                orderItem.getQuantity(),
                orderItem.getCancelled()
        );
    }

    public long getId() {
        return id;
    }

    public long getProduct() {
        return product;
    }

    public int getQuantity() {
        return quantity;
    }

    public int getCancelled() {
        return cancelled;
    }
}

File: argentum-ui/src/app/common/rest-service/response/order-item-response.ts
import { OrderItem } from '../../model/order-item';
import { Product } from '../../model/product';

export class OrderItemResponse {
  id: number;
  product: number;
  quantity: number;
  cancelled: number;
}

export function toOrderItem(response: OrderItemResponse, products: Map<number, Product>): OrderItem {
  return new OrderItem(
    response.id,
    products[response.product],
    response.quantity,
    response.cancelled
  );
}
